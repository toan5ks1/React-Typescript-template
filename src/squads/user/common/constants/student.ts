import { convertEnumKeys } from "src/common/constants/helper";

import { StudentEnrollmentStatus } from "manabuf/usermgmt/v2/enums_pb";

export const KeyStudentEnrollmentStatus = convertEnumKeys(StudentEnrollmentStatus);

export const StudentNewEnrollmentStatusList = [
    "STUDENT_ENROLLMENT_STATUS_WITHDRAWN",
    "STUDENT_ENROLLMENT_STATUS_GRADUATED",
    "STUDENT_ENROLLMENT_STATUS_LOA",
];

export enum StudentKeys {
    RELATIONSHIP = "relationship",
    STUDENTS_STATUS = "studentsEnrollmentStatus",
    ADD_PARENT = "addParent",
    STUDENT_PACKAGES = "studentPackages",
    ADD_STUDENT = "addStudent",
}
