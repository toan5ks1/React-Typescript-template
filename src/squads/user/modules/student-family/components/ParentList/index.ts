export { default } from "./ParentList";
export * from "./ParentList";
