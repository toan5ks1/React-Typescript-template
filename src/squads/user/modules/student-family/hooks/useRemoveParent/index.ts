export * from "./useRemoveParent";
export { default } from "./useRemoveParent";
