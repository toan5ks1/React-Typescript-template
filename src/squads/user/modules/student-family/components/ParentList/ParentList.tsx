import { Box } from "@mui/material";
import Loading from "src/components/Loading";
import PaperRoundedBorders from "src/components/Papers/PaperRoundedBorders";
import WrapperNoData from "src/components/Wrappers/WrapperNoData";

import ParentItem from "../ParentItem";

import { StudentParentDataType } from "src/squads/user/hooks/useParentMapStudent";
import useTranslate from "src/squads/user/hooks/useTranslate";

export interface ParentListProps {
    studentId: string;
    parents: StudentParentDataType[];
    isLoading?: boolean;
    refetchParents: () => void;
}

const ParentList = ({ studentId, parents, isLoading, refetchParents }: ParentListProps) => {
    const t = useTranslate();

    if (isLoading) return <Loading />;

    return (
        <Box data-testid="ParentList__root">
            {parents && parents.length ? (
                parents.map((parent, index) => (
                    <ParentItem
                        key={parent.parent_id}
                        studentId={studentId}
                        parents={parents}
                        parent={parent}
                        index={index}
                        refetchParents={refetchParents}
                    />
                ))
            ) : (
                <PaperRoundedBorders>
                    <WrapperNoData noDataMessage={t("ra.message.noDataInformation")} />
                </PaperRoundedBorders>
            )}
        </Box>
    );
};

export default ParentList;
