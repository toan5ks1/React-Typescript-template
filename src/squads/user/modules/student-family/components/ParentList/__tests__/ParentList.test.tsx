import { createMockListParent } from "src/squads/user/test-utils/mocks/student";
import { TestCommonAppProvider } from "src/squads/user/test-utils/providers";

import ParentList, { ParentListProps } from "../ParentList";

import { render } from "@testing-library/react";
import useRemoveParent from "src/squads/user/modules/student-family/hooks/useRemoveParent";

jest.mock("src/squads/user/hooks/useReissueUserPassword");
jest.mock("src/squads/user/modules/student-family/hooks/useRemoveParent", () => jest.fn());
jest.mock("src/squads/user/hooks/useTranslate");

const mockDataParents = createMockListParent({ length: 2, student_id: "student_id" });

const defaultProps: ParentListProps = {
    studentId: "student_1",
    parents: mockDataParents,
    refetchParents: jest.fn(),
};

const renderComponent = (overrideProps?: ParentListProps) => {
    const props = overrideProps || defaultProps;

    (useRemoveParent as jest.Mock).mockImplementation(() => jest.fn());

    return render(
        <TestCommonAppProvider>
            <ParentList {...props} />
        </TestCommonAppProvider>
    );
};

describe("<ParentList />", () => {
    it("should match snapshot", () => {
        const wrapper = renderComponent();

        expect(wrapper.container).toMatchSnapshot();
    });

    it("should render correct parent item", () => {
        const wrapper = renderComponent();

        expect(wrapper.queryByTestId("ParentList__root")).toBeInTheDocument();
        expect(wrapper.getAllByTestId("ParentItem__root")).toHaveLength(mockDataParents.length);
    });

    it("render with empty parents", () => {
        const props: ParentListProps = {
            studentId: "student_1",
            parents: [],
            refetchParents: jest.fn(),
        };

        const wrapper = renderComponent(props);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.queryByTestId("NoData__message")).toHaveTextContent(
            "ra.message.noDataInformation"
        );
    });
});
