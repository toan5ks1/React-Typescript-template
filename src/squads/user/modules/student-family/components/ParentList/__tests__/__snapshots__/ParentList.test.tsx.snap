// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParentList /> render with empty parents 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-1 {
  background-color: #fff;
  color: #212121;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: none;
}

.emotion-2 {
  border: 1px solid #E0E0E0;
  border-radius: 4px;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
  margin: auto;
  color: #757575;
  background-color: #FAFAFA;
  height: 80px;
  border: none;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-6 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  height: 24px;
  margin-right: 5px;
  color: #00000061;
}

.emotion-7 {
  margin: 0;
  font-size: 0.875rem;
  font-weight: 400;
  letter-spacing: 0.15px;
  text-transform: none;
  font-family: Roboto,sans-serif;
  line-height: 1.43;
}

<body>
    <div>
      <div
        class="MuiBox-root emotion-0"
        data-testid="ParentList__root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 emotion-1"
        >
          <div
            class="MuiBox-root emotion-2"
          >
            <div
              class="MuiBox-root emotion-3"
            >
              <div
                class="MuiBox-root emotion-0"
              >
                <div
                  class="MuiBox-root emotion-5"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-6"
                    data-testid="NoData__icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 15h7v2H7zm0-4h10v2H7zm0-4h10v2H7zm12-4h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-.14 0-.27.01-.4.04-.39.08-.74.28-1.01.55-.18.18-.33.4-.43.64-.1.23-.16.49-.16.77v14c0 .27.06.54.16.78s.25.45.43.64c.27.27.62.47 1.01.55.13.02.26.03.4.03h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7-.25c.41 0 .75.34.75.75s-.34.75-.75.75-.75-.34-.75-.75.34-.75.75-.75zM19 19H5V5h14v14z"
                    />
                  </svg>
                  <p
                    class="MuiTypography-root MuiTypography-body2 emotion-7"
                    data-testid="NoData__message"
                  >
                    ra.message.noDataInformation
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-1 {
  background-color: #fff;
  color: #212121;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: none;
}

.emotion-2 {
  border: 1px solid #E0E0E0;
  border-radius: 4px;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
  margin: auto;
  color: #757575;
  background-color: #FAFAFA;
  height: 80px;
  border: none;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-6 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  height: 24px;
  margin-right: 5px;
  color: #00000061;
}

.emotion-7 {
  margin: 0;
  font-size: 0.875rem;
  font-weight: 400;
  letter-spacing: 0.15px;
  text-transform: none;
  font-family: Roboto,sans-serif;
  line-height: 1.43;
}

<div>
    <div
      class="MuiBox-root emotion-0"
      data-testid="ParentList__root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 emotion-1"
      >
        <div
          class="MuiBox-root emotion-2"
        >
          <div
            class="MuiBox-root emotion-3"
          >
            <div
              class="MuiBox-root emotion-0"
            >
              <div
                class="MuiBox-root emotion-5"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-6"
                  data-testid="NoData__icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 15h7v2H7zm0-4h10v2H7zm0-4h10v2H7zm12-4h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-.14 0-.27.01-.4.04-.39.08-.74.28-1.01.55-.18.18-.33.4-.43.64-.1.23-.16.49-.16.77v14c0 .27.06.54.16.78s.25.45.43.64c.27.27.62.47 1.01.55.13.02.26.03.4.03h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7-.25c.41 0 .75.34.75.75s-.34.75-.75.75-.75-.34-.75-.75.34-.75.75-.75zM19 19H5V5h14v14z"
                  />
                </svg>
                <p
                  class="MuiTypography-root MuiTypography-body2 emotion-7"
                  data-testid="NoData__message"
                >
                  ra.message.noDataInformation
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<ParentList /> should match snapshot 1`] = `
.emotion-1 {
  margin-bottom: 24px;
}

.emotion-2 {
  padding-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-3 {
  margin: 0;
  font-size: 1rem;
  font-weight: 400;
  letter-spacing: 0.15px;
  text-transform: none;
  font-family: Roboto,sans-serif;
  line-height: 1.75;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-5 .IconButtonWithAnchorMenu-sizeSmall {
  padding: 10px;
}

.emotion-5.IconButtonWithAnchorMenu-root:hover {
  color: #2196F3;
}

.emotion-5 .IconButtonWithAnchorMenu-iconButton {
  border-radius: 8px;
  background-color: #EEEEEE;
}

.emotion-6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  padding: 12px;
  font-size: 1.75rem;
  width: 40px;
  height: 40px;
}

.emotion-6::-moz-focus-inner {
  border-style: none;
}

.emotion-6.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-6 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-6:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-6:hover {
    background-color: transparent;
  }
}

.emotion-6.Mui-disabled {
  background-color: transparent;
  color: #BDBDBD;
}

.emotion-7 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.25rem;
}

.emotion-8 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding-top: 8px;
  padding-bottom: 8px;
}

.emotion-11 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-12 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  word-break: break-word;
}

.emotion-13 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 25%;
  -ms-flex-preferred-size: 25%;
  flex-basis: 25%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 25%;
}

@media (min-width:600px) {
  .emotion-13 {
    -webkit-flex-basis: 25%;
    -ms-flex-preferred-size: 25%;
    flex-basis: 25%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 25%;
  }
}

@media (min-width:960px) {
  .emotion-13 {
    -webkit-flex-basis: 25%;
    -ms-flex-preferred-size: 25%;
    flex-basis: 25%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 25%;
  }
}

@media (min-width:1440px) {
  .emotion-13 {
    -webkit-flex-basis: 25%;
    -ms-flex-preferred-size: 25%;
    flex-basis: 25%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 25%;
  }
}

@media (min-width:1920px) {
  .emotion-13 {
    -webkit-flex-basis: 25%;
    -ms-flex-preferred-size: 25%;
    flex-basis: 25%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 25%;
  }
}

.emotion-14 {
  margin: 0;
  font-size: 0.75rem;
  font-weight: 400;
  letter-spacing: 0.4px;
  text-transform: none;
  font-family: Roboto,sans-serif;
  line-height: 1.66;
  color: #757575;
}

.emotion-15 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 66.666667%;
  -ms-flex-preferred-size: 66.666667%;
  flex-basis: 66.666667%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 66.666667%;
}

@media (min-width:600px) {
  .emotion-15 {
    -webkit-flex-basis: 66.666667%;
    -ms-flex-preferred-size: 66.666667%;
    flex-basis: 66.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 66.666667%;
  }
}

@media (min-width:960px) {
  .emotion-15 {
    -webkit-flex-basis: 66.666667%;
    -ms-flex-preferred-size: 66.666667%;
    flex-basis: 66.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 66.666667%;
  }
}

@media (min-width:1440px) {
  .emotion-15 {
    -webkit-flex-basis: 66.666667%;
    -ms-flex-preferred-size: 66.666667%;
    flex-basis: 66.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 66.666667%;
  }
}

@media (min-width:1920px) {
  .emotion-15 {
    -webkit-flex-basis: 66.666667%;
    -ms-flex-preferred-size: 66.666667%;
    flex-basis: 66.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 66.666667%;
  }
}

.emotion-17 {
  margin: 0;
  font-size: 0.875rem;
  font-weight: 400;
  letter-spacing: 0.15px;
  text-transform: none;
  font-family: Roboto,sans-serif;
  line-height: 1.43;
}

<div>
  <div
    class="MuiBox-root emotion-0"
    data-testid="ParentList__root"
  >
    <div
      class="MuiBox-root emotion-1"
      data-testid="ParentItem__root"
      data-value="parent_id 0"
    >
      <div
        class="MuiBox-root emotion-2"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 emotion-3"
          data-testid="ParentItem__title"
        >
          resources.students_erp.titles.parent
           #
          1
        </h6>
        <div
          class="MuiBox-root emotion-4"
          data-testid="ActionPanel__root"
        >
          <div
            class="IconButtonWithAnchorMenu-root emotion-5"
          >
            <button
              aria-haspopup="menu"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge IconButtonWithAnchorMenu-root emotion-6"
              data-testid="ActionPanel__trigger"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall emotion-7"
                data-testid="MoreHorizIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root emotion-8"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root emotion-0"
        data-testid="ParentItem__info"
      >
        <div
          class="MuiBox-root emotion-10"
        >
          <div
            class="MuiGrid-root MuiGrid-container emotion-11"
            data-testid="ParentItem__infoItem"
          >
            <div
              class="MuiGrid-root MuiGrid-container emotion-12"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 emotion-13"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption emotion-14"
                >
                  resources.students_erp.labels.name
                </span>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 emotion-15"
                data-testid="ParentItem__Name"
              >
                <div
                  class="MuiBox-root emotion-4"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 emotion-17"
                  >
                    Parent Name 0
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container emotion-11"
            data-testid="ParentItem__infoItem"
          >
            <div
              class="MuiGrid-root MuiGrid-container emotion-12"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 emotion-13"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption emotion-14"
                >
                  resources.students_erp.labels.relationship
                </span>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 emotion-15"
                data-testid="ParentItem__Relationship"
              >
                <div
                  class="MuiBox-root emotion-4"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 emotion-17"
                  >
                    resources.students_erp.choices.relationship.1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root emotion-10"
        >
          <div
            class="MuiGrid-root MuiGrid-container emotion-11"
            data-testid="ParentItem__infoItem"
          >
            <div
              class="MuiGrid-root MuiGrid-container emotion-12"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 emotion-13"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption emotion-14"
                >
                  resources.students_erp.labels.phoneNumber
                </span>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 emotion-15"
                data-testid="ParentItem__PhoneNumber"
              >
                <div
                  class="MuiBox-root emotion-4"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 emotion-17"
                  >
                    (+81) Phone Number 0
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container emotion-11"
            data-testid="ParentItem__infoItem"
          >
            <div
              class="MuiGrid-root MuiGrid-container emotion-12"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 emotion-13"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption emotion-14"
                >
                  resources.students_erp.labels.email
                </span>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 emotion-15"
                data-testid="ParentItem__Email"
              >
                <div
                  class="MuiBox-root emotion-4"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 emotion-17"
                  >
                    parent_email0@example.com
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root emotion-1"
      data-testid="ParentItem__root"
      data-value="parent_id 1"
    >
      <div
        class="MuiBox-root emotion-2"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 emotion-3"
          data-testid="ParentItem__title"
        >
          resources.students_erp.titles.parent
           #
          2
        </h6>
        <div
          class="MuiBox-root emotion-4"
          data-testid="ActionPanel__root"
        >
          <div
            class="IconButtonWithAnchorMenu-root emotion-5"
          >
            <button
              aria-haspopup="menu"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge IconButtonWithAnchorMenu-root emotion-6"
              data-testid="ActionPanel__trigger"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall emotion-7"
                data-testid="MoreHorizIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root emotion-8"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root emotion-0"
        data-testid="ParentItem__info"
      >
        <div
          class="MuiBox-root emotion-10"
        >
          <div
            class="MuiGrid-root MuiGrid-container emotion-11"
            data-testid="ParentItem__infoItem"
          >
            <div
              class="MuiGrid-root MuiGrid-container emotion-12"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 emotion-13"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption emotion-14"
                >
                  resources.students_erp.labels.name
                </span>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 emotion-15"
                data-testid="ParentItem__Name"
              >
                <div
                  class="MuiBox-root emotion-4"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 emotion-17"
                  >
                    Parent Name 1
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container emotion-11"
            data-testid="ParentItem__infoItem"
          >
            <div
              class="MuiGrid-root MuiGrid-container emotion-12"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 emotion-13"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption emotion-14"
                >
                  resources.students_erp.labels.relationship
                </span>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 emotion-15"
                data-testid="ParentItem__Relationship"
              >
                <div
                  class="MuiBox-root emotion-4"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 emotion-17"
                  >
                    resources.students_erp.choices.relationship.1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root emotion-10"
        >
          <div
            class="MuiGrid-root MuiGrid-container emotion-11"
            data-testid="ParentItem__infoItem"
          >
            <div
              class="MuiGrid-root MuiGrid-container emotion-12"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 emotion-13"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption emotion-14"
                >
                  resources.students_erp.labels.phoneNumber
                </span>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 emotion-15"
                data-testid="ParentItem__PhoneNumber"
              >
                <div
                  class="MuiBox-root emotion-4"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 emotion-17"
                  >
                    (+81) Phone Number 1
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container emotion-11"
            data-testid="ParentItem__infoItem"
          >
            <div
              class="MuiGrid-root MuiGrid-container emotion-12"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 emotion-13"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption emotion-14"
                >
                  resources.students_erp.labels.email
                </span>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 emotion-15"
                data-testid="ParentItem__Email"
              >
                <div
                  class="MuiBox-root emotion-4"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 emotion-17"
                  >
                    parent_email1@example.com
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
